### PR TITLE
add swsscommon python library

### PR DESCRIPTION
**Pull request by lguohan**
```
lgh@0c825b457549:/data/sonic/sonic-swss-common/tests$ /usr/local/bin/pytest 
================================================== test session starts ===================================================
platform linux2 -- Python 2.7.9, pytest-3.2.3, py-1.4.34, pluggy-0.4.0
rootdir: /data/sonic/sonic-swss-common/tests, inifile:
plugins: cov-1.8.0
collected 1 item                                                                                                          

test_redis_ut.py .

==================================================== warnings summary ====================================================
None
  pytest_funcarg__cov: declaring fixtures using "pytest_funcarg__" prefix is deprecated and scheduled to be removed in py.

-- Docs: http://doc.pytest.org/en/latest/warnings.html
========================================== 1 passed, 1 warnings in 0.03 seconds ==========================================
```

### CODE REVIEW

**Comment by lguohan** [comment]
retest this please

**Comment by lguohan** [comment]
depends on #154

File: pyext/swsscommon.i [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#include "redisselect.h"
+#include "redistran.h"
+#include "producerstatetable.h"
+#include "consumertablebase.h"
+#include "consumerstatetable.h"
+#include "subscriberstatetable.h"
+#include "notificationconsumer.h"
+%}
+
+#include "schema.h"
```
Comment (jleveque) [inline pyext/swsscommon.i]: `%include`

Comment (lguohan) [reply pyext/swsscommon.i]: done

File: common/consumertablebase.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
+#include "consumertablebase.h"
+
+namespace swss {
+
+ConsumerTableBase::ConsumerTableBase(DBConnector *db, std::string tableName, int popBatchSize):
```
Comment (qiluo-msft) [inline common/consumertablebase.cpp]: >ConsumerTableBase [](start = 19, length = 17)

This file would be covered by another PR.

File: common/redispipeline.h [hunk 3]
```diff
@@ -11,7 +11,7 @@ namespace swss {
 class RedisPipeline {
 public:
     const size_t COMMAND_MAX;
-    const static int NEWCONNECTOR_TIMEOUT = 0;
+    static constexpr int NEWCONNECTOR_TIMEOUT = 0;
```
Comment (qiluo-msft) [inline common/redispipeline.h]: >constexpr [](start = 11, length = 9)

What is wrong with original code? Just curios?

File: pyext/Makefile.am [hunk 4]
```diff
@@ -0,0 +1,12 @@
+SWIG_SOURCES = swsscommon.i
+
+pkgpython_PYTHON = swsscommon.py __init__.py
+pkgpyexec_LTLIBRARIES = _swsscommon.la
```
Comment (qiluo-msft) [inline pyext/Makefile.am]: >_ [](start = 24, length = 1)

Is there a name convention here?

File: tests/test_redis_ut.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+from swsscommon import swsscommon
```
Comment (qiluo-msft) [inline tests/test_redis_ut.py]: Can we give a hint what python version is needed?

Comment (lguohan) [reply tests/test_redis_ut.py]: currently 2.7, I haven't compiled the swig to support python3 yet.

File: pyext/Makefile.am (hunk 4 above)
Comment (lguohan) [reply pyext/Makefile.am]: yes, this is for python submodule, if the python submodule name is A, then the c++ so is _A.

File: common/consumertablebase.cpp (hunk 2 above)
Comment (lguohan) [reply common/consumertablebase.cpp]: this is in #154, I'll rebase once that one is merged.

File: common/redispipeline.h (hunk 3 above)
Comment (lguohan) [reply common/redispipeline.h]: swig compile error. I did not look into it deeper.

Comment (jleveque) [reply common/redispipeline.h]: Might be related to this: https://github.com/swig/swig/issues/241
